src/renate/benchmark/experiment_config.py

```diff
@@ -398,3 +405,13 @@ def lr_scheduler_fn(
 
 def metrics_fn(num_outputs: int) -> Dict:
     return {"accuracy": MulticlassAccuracy(num_classes=num_outputs, average="micro")}
+
+
+def optimizer_fn(
+    optimizer: str,
+    learning_rate: float,
+    weight_decay: float,
+    momentum: float = 0.0,  # TODO: fix problem that occurs when removing this
+) -> Callable:
+    if optimizer == "AdamW":
+        return partial(AdamW, lr=learning_rate, weight_decay=weight_decay)
```

Comment:
can this written as: 
`partial(getattr(torch.optim, optimizer),  lr=learning_rate, weight_decay=weight_decay)`? Why the specific handling of AdamW?

Reply:
in principle there is no problem if we make it more general. however, how would that work with SGD + momentum? right now, this logic will only be triggered for AdamW and otherwise we will fall back to the standard optimizers (SGD, Adam), we have in the library.